pkg/core/runtime/reports/reports.go

```diff
@@ -65,10 +65,26 @@ func fetchZones(rt core_runtime.Runtime) (*system.ZoneResourceList, error) {
 	if err := rt.ReadOnlyResourceManager().List(context.Background(), &zones); err != nil {
 		return nil, errors.Wrap(err, "could not fetch zones")
 	}
-
 	return &zones, nil
 }
 
+func fetchNumOfServices(rt core_runtime.Runtime) (int, int, error) {
+	insights := mesh.ServiceInsightResourceList{}
+	if err := rt.ReadOnlyResourceManager().List(context.Background(), &insights); err != nil {
+		return -1, -1, errors.Wrap(err, "could not fetch service insights")
```

Comment:
I think the common rule is if an error is not nil, then other return values don't make sense. That's why I believe `return 0, 0, errors.Wrap(...)` should be fine